src/Reflection/ReflectionClass.php

```diff
@@ -338,20 +338,33 @@ private function getMethodsIndexedByName() : array
     /**
      * Fetch an array of all methods for this class.
      *
+     * @param ?int $filter
      * @return ReflectionMethod[]
      */
-    public function getMethods() : array
+    public function getMethods(?int $filter = null) : array
     {
-        return array_values($this->getMethodsIndexedByName());
+        if (null === $filter) {
```

Comment:
Thanks @Ocramius for your great explanation :)

But I've few remarks:
- `is_null()` does the job in C, and the comparison is not the same things. In fact is it really important to optimize this kind of things? (This is a real question :smile: )
- You talking about big O notation, is it really necessary to think to this low level ? But I do a [benchmark](https://gist.github.com/roukmoute/a65eb6749a8f506ac048b008f07dc27a), and finally with latest version of PHP, `is_null` is more fastest.

Here is a small [screencast](https://asciinema.org/a/08670iigxidrc3okzftxrijsq) about that.